### PR TITLE
Tl/base logger

### PR DESCRIPTION
**Pull request by tblenahan**
This PR contains additions/updates to the BaseLogger struct and the methods attached to it.

Submit successes and failures tracked as int64 types by the BaseLogger and incremented atomically during submit execution.

Submit method implemented which will conduct the final step in the logging process submitting to the desire Resurface database instance specified by the URL attached to the BaseLogger (when initialized).

The submit method needs testing for 100% functionality confirmation but at this point, no compilation errors should occur.

### CODE REVIEW

File: src/logger/BaseLogger.go [hunk 1]
```diff
@@ -79,39 +56,58 @@ func NewBaseLogger(_agent string, _url string, _enabled bool, _queue []string) *
...
-		// url_connection.setReadTimeout(1000);
-		// url_connection.setRequestMethod("POST");
-		// url_connection.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
-		// url_connection.setRequestProperty("User-Agent", "Resurface/" + version + " (" + agent + ")");
-		// url_connection.setDoOutput(true);
-		if obj.SkipCompression() {
-			url_connection.setRequestProperty("Content-Encoding", "deflated")
-		} else {
+		// not 100% sure this works (needs testing)
+		submitRequest := http.NewRequest("POST", logger.url, bytes.NewBuffer([]byte(msg)))
```
Comment (Devinmagnuson) [inline src/logger/BaseLogger.go:79]: I'm guessing this is for building the Request Header without setting all the of different fields that come with it.

File: src/logger/BaseLogger.go [hunk 2]
```diff
@@ -79,39 +56,58 @@ func NewBaseLogger(_agent string, _url string, _enabled bool, _queue []string) *
...
+		submitResponse, err := http.DefaultClient.Do(&submitRequest)
+		if err != nil {
+			atomic.AddInt64(&logger.submitFailures, 1)
+		}
+
+		if submitResponse.StatusCode == 204 {
+			atomic.AddInt64(&logger.submitSuccesses, 1)
+		} else {
+			atomic.AddInt64(&logger.submitFailures, 1)
+		}
```
Comment (Devinmagnuson) [inline src/logger/BaseLogger.go:110]: I like the implementation with flate or deflate and the translating from Java.  For the submitFailures or submitSuccesses is do we output these values anywhere to track or so far we just store if it passes or fails?

File: src/logger/HttpLogger.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	for i, val := range msg {
+		stringified += "[" + strings.Join(val, ", ") + "]"
+		if i != n-1 {
+			stringified += ","
+		}
+	}
+	stringified = "[" + stringified + "]"
+
+	return stringified
+}
```
Comment (Devinmagnuson) [inline src/logger/HttpLogger.go:61]: This is a curious implementation and I wonder if I can re-use it inside the parseable function in logger helper

**Review by Devinmagnuson — APPROVED** [review]
Looks good to me.  A little bias since I worked on parts of it but I'm sure if anything doesn't sit right we'll find out quickly when we integrate.

**Review by KyleNeubarth — APPROVED** [review]

**Pull request merged**